cpp/src/arrow/compute/expression.cc

```diff
@@ -1331,7 +1331,13 @@ struct Inequality {
       return call->function_name == "is_valid" ? literal(true) : literal(false);
     }
 
-    if (call->function_name == "is_in") {
+    // The maximum number of values in the expression set of values
```

Comment:
Why not do this is `SimplifyIsIn` instead? Also, can you add a GH issue reference?

Reply:
Agreed. We tend to hide rule-specific logic within `SimplifyRule` function.